test/suites/integration/bsp/volunteer.test.ts

```diff
@@ -233,26 +233,36 @@ describeBspNet("Multiple BSPs volunteer ", ({ before, createBspApi, createUserAp
 
     await userApi.sealBlock(txs, shUser);
 
-    //There should be pending extrinsics for all files from BSP (volunteer)
-    await userApi.assert.extrinsicPresent({
-      module: "fileSystem",
-      method: "bspVolunteer",
-      checkTxPool: true,
-      assertLength: source.length
+    // Get the new storage request events, making sure we have 3
+    const storageRequestEvents = await userApi.assert.eventMany("fileSystem", "NewStorageRequest");
+    strictEqual(storageRequestEvents.length, 3);
+
+    // Get the file keys from the storage request events
+    const fileKeys = storageRequestEvents.map((event) => {
+      const dataBlob =
+        userApi.events.fileSystem.NewStorageRequest.is(event.event) && event.event.data;
+      if (!dataBlob) {
+        throw new Error("Event doesn't match Type");
+      }
+      return dataBlob.fileKey;
     });
 
-    await userApi.sealBlock();
+    // Wait for the BSP to volunteer
+    await userApi.wait.bspVolunteer(source.length);
 
-    await sleep(5000); // wait for the bsp to download the files
-    await userApi.assert.extrinsicPresent({
-      module: "fileSystem",
-      method: "bspConfirmStoring",
-      checkTxPool: true,
-      assertLength: 1,
-      timeout: 10000
-    });
+    // Wait for the BSP to receive and store all files
+    for (let i = 0; i < source.length; i++) {
+      const fileKey = fileKeys[i];
+      await bspApi.assert.log({
```

Comment:
I don't think we should depend on logs. I would use the rpc method `isFileInFileStorage`, that way we don't need to maintain log outputs.

Reply:
Changed logs for a specific wait function that waits to get `FileFound` from the `isFileInFileStorage` RPC call